ui/static/ui/js/learning_resources.jsx

```diff
@@ -390,6 +415,30 @@ define('learning_resources',
         thiz.setState({loaded: true});
       });
     },
+    componentWillReceiveProps: function(nextProps) {
+      var unSavedChangesMessage = "You have unsaved changes. " +
+        "Please save and exit.";
+
+      if (!_.eq(this.state.description, this.state.descriptionCopy)) {
```

Comment:
We should do this comparison at the point that the state changes, `handleDescription` and all of the functions that update the terms.